src/articulation.js

```diff
@@ -56,126 +139,75 @@ export class Articulation extends Modifier {
+      articulation: { between_lines: canSitBetweenLines },
+      text_line: textLine,
+      context: ctx,
+    } = this;
+
+    if (!ctx) {
       throw new Vex.RERR('NoContext', "Can't draw Articulation without a context.");
     }
-    if (!(this.note && (this.index !== null))) {
+
+    if (!note || index == null) {
       throw new Vex.RERR('NoAttachedNote', "Can't draw Articulation without a note and index.");
     }
 
-    const stem_direction = this.note.getStemDirection();
-    const stave = this.note.getStave();
-
-    const is_on_head = (this.position === Modifier.Position.ABOVE &&
-                      stem_direction === StaveNote.STEM_DOWN) ||
-                     (this.position === Modifier.Position.BELOW &&
-                      stem_direction === StaveNote.STEM_UP);
-
-    const needsLineAdjustment = (articulation, note_line, line_spacing) => {
-      const offset_direction = (articulation.position === Modifier.Position.ABOVE) ? 1 : -1;
-      const duration = articulation.getNote().getDuration();
-      if (!is_on_head && Flow.durationToNumber(duration) <= 1) {
-        // Add stem length, unless it's on a whole note.
-        note_line += offset_direction * 3.5;
-      }
-
-      const articulation_line = note_line + (offset_direction * line_spacing);
-
-      if (articulation_line >= 1 &&
-         articulation_line <= 5 &&
-         articulation_line % 1 === 0) {
-        return true;
-      }
-
-      return false;
-    };
+    const stave = note.getStave();
+    const staffSpace = stave.getSpacingBetweenLines();
+    const isTab = note.getCategory() === 'tabnotes';
 
     // Articulations are centered over/under the note head.
-    const start = this.note.getModifierStartXY(this.position, this.index);
-    let glyph_y = start.y;
-    let shiftY = 0;
-    let line_spacing = 1;
-    const spacing = stave.getSpacingBetweenLines();
-    const is_tabnote = this.note.getCategory() === 'tabnotes';
-    const stem_ext = this.note.getStem().getExtents();
-
-    let top = stem_ext.topY;
-    let bottom = stem_ext.baseY;
-
-    if (stem_direction === StaveNote.STEM_DOWN) {
-      top = stem_ext.baseY;
-      bottom = stem_ext.topY;
-    }
-
-    // TabNotes don't have stems attached to them. Tab stems are rendered
-    // outside the stave.
-    if (is_tabnote) {
-      if (this.note.hasStem()) {
-        if (stem_direction === StaveNote.STEM_UP) {
-          bottom = stave.getYForBottomText(this.text_line - 2);
-        } else if (stem_direction === StaveNote.STEM_DOWN) {
-          top = stave.getYForTopText(this.text_line - 1.5);
-        }
-      } else { // Without a stem
-        top = stave.getYForTopText(this.text_line - 1);
-        bottom = stave.getYForBottomText(this.text_line - 2);
-      }
+    const { x } = note.getModifierStartXY(position, index);
+    const shouldSitOutsideStaff = !canSitBetweenLines || isTab;
+
+    const y = {
+      [ABOVE]: () => {
+        glyph.setOrigin(0.5, 1);
```

Comment:
@0xfe tried linking to this in my previous comment but it doesn't seem to anchor to the right point